events/giveaways/endedGiveawayReactionAdded.js

```diff
@@ -1,7 +1,28 @@
+const { MessageEmbed, MessageButton, MessageActionRow } = require("discord.js")
 const Discord = require("discord.js")
+
 module.exports = {
-  async execute(giveaway, member, reaction){
+  async execute(trade, member, reaction){
      reaction.users.remove(member.user);
-  member.send(`**Aw snap! Looks Like that giveaway has already ended!**`).catch(e => {})
+  member.send({  
+    components: [new MessageActionRow()
+                .addComponents(
+                      new MessageButton()
+                  .setLabel("See the shift trade message.")
+                  .setStyle("LINK")
+                  .setURL(`https://discord.com/channels/${trade.guildId}/${trade.channelId}/${trade.messageId}`)
+                  .setEmoji('973495590921043968'))],
```

Comment:
FYI you can add emojis directly into the source code. Look eg [here](https://github.com/lighthouse-labs/rudder-rails/blob/e0573a57e73699eb440e3592e2db23bc2a75a079/discord-bot/src/utils/emoji-selector.ts)

Reply:
I assumed, just how the OG author had it set though so I decided to leave it for now.